### PR TITLE
DrivAerML: 4L/256d+T_max=10+lr=3e-4 — compound config

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The DrivAerML baseline (val=12.96%) used AdamW lr=5e-4 with T_max=30. Two separate insights suggest we can do better with a compound change:

1. **lr=3e-4 vs 5e-4**: TandemFoil's winning config uses lr=3e-4 with Lion. For AdamW on DrivAerML, a slightly lower lr=3e-4 may allow more precise convergence in the sharp loss landscape of this complex 3D dataset.

2. **T_max=10 vs T_max=30**: TandemFoil's best result uses T_max=10, which creates very frequent cosine troughs (~75 cycles/epoch at 750 steps/epoch). For DrivAerML at ~394 steps/epoch, T_max=10 gives ~39 cycles/epoch — still far more aggressive than T_max=30's ~13 cycles/epoch. More frequent warm restarts may help escape local minima faster in this high-dimensional surface prediction problem.

We keep the proven 4L/256d architecture (the key innovation from PR #2550) and test whether the lower lr + more aggressive T_max compound to further improve surface pressure prediction.

## Instructions

Run the following command exactly:

```bash
cd target/icml2026 && python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 3e-4 \
  --cosine_t_max 10 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --wandb_group drivaerml-4L256d-compound
```

**What to watch for:** Compare convergence speed vs the baseline (W&B: 8s5i8y06). With T_max=10 and lower lr, we expect smoother, more stable training. If val loss plateaus above 12.96% by epoch 60, the experiment can be considered conclusive.

**Note:** The `--batch-size 1` and surface point limits are required to prevent OOM on DrivAerML. Do not remove these flags.

## Baseline

Current best DrivAerML result (PR #2550):

| Metric | Value |
|--------|-------|
| val_primary/surface_rel_l2_pct | **12.96%** |
| test_primary/surface_rel_l2_pct | 14.41% |
| Best epoch | 43 |
| W&B run | [8s5i8y06](https://wandb.ai/wandb/senpai/runs/8s5i8y06) |

**Config:** AdamW lr=5e-4, cosine T_max=30, Fourier, no-EMA, 4L/256d, batch-size=1, surface-points=50000

**Target:** val_primary/surface_rel_l2_pct < 12.96%

**Reproduce baseline:**
```bash
cd target/icml2026 && python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine_t_max 30 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200
```

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT norman:

## Results

### Compound config: lr=3e-4 + T_max=10

| Metric | Baseline (PR #2550) | This experiment | Delta |
|--------|-------------------|-----------------|-------|
| val_primary/surface_rel_l2_pct | **12.96%** | 14.90% (best, ep 41) | +1.94% (worse) |
| test_primary/surface_rel_l2_pct | 14.41% | 15.10% | +0.69% (worse) |
| Best epoch | 43 | 41 (of 44) | — |

**W&B run:** [l2kaq446](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/l2kaq446)
**Epochs:** 44 (30-min timeout)

**Command:**
```bash
cd target/icml2026 && python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 3e-4 \
  --cosine-t-max 10 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --wandb-group drivaerml-4L256d-compound \
  --agent norman \
  --wandb-name "norman/drivaerml-4L256d-tmax10-lr3e4"
```

### What happened

**The compound change did not beat the baseline.** At comparable epoch counts (44 vs 43), lr=3e-4 + T_max=10 achieved best val=14.90% vs the baseline's 12.96% — a 15% relative degradation.

**Key observations:**

1. **Lower LR slowed convergence.** The lr=3e-4 run was consistently behind the baseline trajectory at matched epoch counts. Training loss was still dropping (0.046 at epoch 44), suggesting the model was still learning but at a slower rate.

2. **T_max=10 produced high-amplitude val oscillations.** With T_max=10 at 394 steps/epoch, the cosine scheduler completes ~39.4 cycles per epoch. Val metrics swung widely within each cycle (e.g., epoch 41=14.90%, epoch 42=16.79%, epoch 43=17.45%, epoch 44=15.06%). The baseline's T_max=30 (~13 cycles/epoch) had smoother convergence.

3. **Trough-to-trough progression shows diminishing returns.** LR=0 trough points: ep5=33.96%, ep15=23.39%, ep25=19.31%, ep35=17.37%. The improvement rate decelerated, and trough values didn't consistently produce the best metrics (ep32 trough=19.02% was worse than ep31=17.62%).

4. **TandemFoil insights don't transfer directly to DrivAerML.** TandemFoil benefits from very aggressive restarts (T_max=10) with Lion optimizer at lr=3e-4, but DrivAerML's higher-dimensional surface prediction problem appears to need the higher lr=5e-4 for sufficient gradient signal and moderate T_max=30 for smoother optimization.

### Suggested follow-ups

- **lr=5e-4 + T_max=10** — isolate whether T_max=10 alone helps when paired with the baseline's higher LR. This would test the restart frequency hypothesis independently.
- **lr=3e-4 + T_max=30** — isolate whether the lower LR helps with the baseline's T_max. This would test the lr hypothesis independently.
- **Longer training** — the run was still converging (loss=0.046, still dropping). With a longer budget, lr=3e-4 + T_max=10 might eventually catch up, though the deceleration pattern suggests it would plateau above the baseline.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2603 because ADVISOR: Dead end — T_max=10+lr=3e-4 compound on DrivAerML gives 14.90% (worse than 12.70% baseline). T_max=10 creates too-frequent restarts for DrivAerML's complex 3D surface (high variance, unstable troughs). lr=3e-4 too slow. TandemFoil hyperparams do not transfer to DrivAerML. Closing..
